site/cds_rdm/views.py

```diff
@@ -60,3 +60,46 @@ def inspire_link_render(record):
             )
         )
     return ret
+
+
+def get_linked_records_search_query(record):
+    """Build search query for linked records.
+
+    Returns a search query string to find:
+    1. Records that this record references in related_identifiers (scheme="cds")
+    2. Records that reference this record in their related_identifiers (scheme="cds")
+    """
+    # Get CDS identifiers from related_identifiers
+    related_identifiers = record.data["metadata"].get("related_identifiers", [])
+    cds_related_ids = [
+        rel_id.get("identifier")
+        for rel_id in related_identifiers
+        if rel_id.get("scheme") == "cds" and rel_id.get("identifier")
+    ]
+
+    # Build query parts
+    query_parts = []
+
+    # Part 1: Records that this record references (forward)
+    # Search by record id using the CDS identifier
+    for cds_id in cds_related_ids:
+        query_parts.append(f'id:"{cds_id}"')
+
+    # Part 2: Records that reference this record (reverse)
+    # Find records that have this record's CDS PIDs in their related_identifiers
+    record_id = record.data.get("id")
+    query_parts.append(
+        f'(metadata.related_identifiers.scheme:"cds" AND '
+        f'metadata.related_identifiers.identifier:"{record_id}")'
+    )
+
+    if not query_parts:
+        return None
+
+    # Combine all query parts with OR
+    combined_query = " OR ".join(query_parts)
+
+    # Exclude the current record and only show published records
+    final_query = f'({combined_query}) AND is_published:true NOT id:"{record_id}"'
+
+    return final_query
```

Comment:
is there any way to find sibling references too? fe
I am on detail page of record A.
I find record B references record A.
there is also record C which references B but not A.
Can I include record C in this search too? Should I? tbd, ping @zzacharo @ntarocco